x-pack/plugins/endpoint/common/types.ts

```diff
@@ -611,47 +611,27 @@ export enum HostPolicyResponseActionStatus {
 }
 
 /**
- * The details of a given action
+ * Host Policy Response Applied Action
  */
-export interface HostPolicyResponseActionDetails {
+export interface HostPolicyResponseAppliedAction {
+  name: string;
```

Comment:
It would be nice if we could have a permissive list of possible actions that we could use here and everywhere else where we do `string[]`. Some like this:

```typescript
type HostPolicyActionName = 
  | 'download_model'
  | 'ingest_events_config'
  | 'workflow'
  | 'configure_elasticsearch_connection'
  | 'configure_kernel'
  | 'configure_logging'
  | 'configure_malware'
  | 'connect_kernel'
  | 'detect_file_open_events'
  | 'detect_file_write_events'
  | 'detect_image_load_events'
  | 'detect_process_events'
  | 'download_global_artifacts'
  | 'load_config'
  | 'load_malware_model'
  | 'read_elasticsearch_config'
  | 'read_events_config'
  | 'read_kernel_config'
  | 'read_logging_config'
  | 'read_malware_config'
  | string;
```
Then on this line (and other areas below):

```suggestion
  name: HostPolicyActionName;
```

Tried it quickly on my IDE and seems to work ok even with custom unknown strings:

![image](https://user-images.githubusercontent.com/56442535/81580075-f55c9500-937a-11ea-9ad6-ada23972f147.png)